model/class.tx_ttproducts_article_base.php

```diff
@@ -473,7 +473,7 @@ public function mergeGraduatedPrice(
     ): void {
     }
 
-    public function getTotalDiscount(&$row, $pid = 0): void
+    public function getTotalDiscount(&$row, $pid = 0)
```

Comment:
A lot of void annotations for functions have been removed. Is this intended`

Reply:
Yes, that's because the parent method, which probably should be an abstract method, doesn't contain a method body and therefore returns void but the child classes actually contain logic and some return something, others don't. The only way to fix this quickly was to remove the return type declaration again.